src/api/AppApi.js

```diff
@@ -0,0 +1,254 @@
+    })
+    .catch((error :Error) => {
+      LOG.error(error);
+      return Promise.reject(error);
+    });
+}
+
+/**
+  * `GET /app/config/{appId}`
+  *
+  * Loads all available configurations for a provided app id
+  *
+  * @static
+  * @memberof lattice.AppApi
+  * @param {UUID} appId
+  * @return {Promise<Schema>} - a Promise that will resolve with all available configurations for an app
+  *
+  * @example
+  * AppApi.getConfigurations("ec6865e6-e60e-424b-a071-6a9c1603d735");
+  */
+export function getConfigurations(appId :UUID) :Promise<*> {
+
+  let errorMsg = '';
+
+  if (!isValidUuid(appId)) {
+    errorMsg = 'invalid parameter: appId must be a valid UUID';
+    LOG.error(errorMsg, appId);
+    return Promise.reject(errorMsg);
+  }
+
+  return getApiAxiosInstance(APP_API)
+    .get(`/${CONFIG_PATH}/${appId}`)
+    .then((axiosResponse) => {
+      return axiosResponse.data;
+    })
+    .catch((error :Error) => {
+      LOG.error(error);
+      return Promise.reject(error);
+    });
+}
+
+/**
+  * `GET /app/{appId}`
+  *
+  * Loads app with provided id
+  *
+  * @static
+  * @memberof lattice.AppApi
+  * @param {UUID[]} appTypeIds
+  * @return {Promise<Schema>} - a Promise that will resolve with the details of an app
+  *
+  * @example
+  * AppApi.getApp("0c8be4b7-0bd5-4dd1-a623-da78871c9d0e");
+  */
+
+export function getApp(appId :UUID) :Promise<*> {
+
+  let errorMsg = '';
+
+  if (!isValidUuid(appId)) {
+    errorMsg = 'invalid parameter: appId must be a valid UUID';
+    LOG.error(errorMsg, appId);
+    return Promise.reject(errorMsg);
+  }
+
+  return getApiAxiosInstance(APP_API)
+    .get(`/${appId}`)
+    .then((axiosResponse) => {
+      return axiosResponse.data;
+    })
+    .catch((error :Error) => {
+      LOG.error(error);
+      return Promise.reject(error);
+    });
+}
+
+/**
+  * `GET /app/{appName}`
+  *
+  * Loads app with provided name
+  *
+  * @static
+  * @memberof lattice.AppApi
+  * @param {UUID[]} appTypeIds
+  * @return {Promise<Schema>} - a Promise that will resolve with the details of an app
+  *
+  * @example
+  * AppApi.getAppByName("bhr");
+  */
+
+export function getAppByName(appName :String) :Promise<*> {
+
+  let errorMsg = '';
+
+  if (!isNonEmptyString(appName)) {
+    errorMsg = 'invalid parameter: appName must be a non-empty string';
+    LOG.error(errorMsg, appName);
+    return Promise.reject(errorMsg);
+  }
+
+  return getApiAxiosInstance(APP_API)
+    .get(`/${LOOKUP_PATH}/${appName}`)
+    .then((axiosResponse) => {
+      return axiosResponse.data;
+    })
+    .catch((error :Error) => {
+      LOG.error(error);
+      return Promise.reject(error);
+    });
+}
+
+/**
+  * `GET /app/type`
+  *
+  * Loads app type objects for provided ids
+  *
+  * @static
+  * @memberof lattice.AppApi
+  * @param {UUID[]} appTypeIds
+  * @return {Promise<Schema>} - a Promise that will resolve with a mapping from app type ids to app types
+  *
+  * @example
+  * AppApi.getAppTypeIds(
```

Comment:
Your parameters are wrong in this example. This function takes an array, but you're passing in each UUID as an individual param.